dockstore-webservice/src/main/java/io/dockstore/webservice/resources/proposedGA4GH/ToolsApiExtendedServiceImpl.java

```diff
@@ -217,6 +219,12 @@ public Response toolsIndexGet(SecurityContext securityContext) {
 
     @Override
     public Response toolsIndexSearch(String query, MultivaluedMap<String, String> queryParameters, SecurityContext securityContext) {
+        String unableToUseESMsg = "Could not use Elasticsearch search";
+        if (!ELASTIC_SEARCH_CONCURRENCY_LIMIT.tryAcquire(1)) {
+            LOG.error(unableToUseESMsg + ": too many concurrent requests.");
+            throw new CustomWebApplicationException("Could not use Elasticsearch search", HttpStatus.SC_INTERNAL_SERVER_ERROR);
+        }
+
         if (!config.getEsConfiguration().getHostname().isEmpty()) {
```

Comment:
It might be better to make this a big try...finally block, and then do the `release(1)` once, in the finally. In case the logic moves around and we end up accidentally not calling it or calling it too often.

Reply:
or that